### PR TITLE
Week 2 project pull request from Maria Sjögren

### PR DESCRIPTION
**Pull request by molbimien**
Here's my pull request for this week's project: Build a news site.

### CODE REVIEW

File: code/style.css [hunk 1]
```diff
@@ -1,4 +1,322 @@
...
+  border-top: 1px solid #e0dfdf;
+  font-size: 0.8rem;
+  text-transform: uppercase;
+}
+
+footer a:link {
+  color: #DB398D;
+}
+
+footer a:visited {
```
Comment (annaester) [inline code/style.css:233]: Nice touch!

File: code/style.css [hunk 2]
```diff
@@ -1,4 +1,322 @@
...
+  color: inherit;
+  text-decoration: none;
+}
+a:before, a:after, a > span:before, a > span:after {
+  content: '';
+  position: absolute;
+  transition: transform .5s ease;
+}
+
+.hover:before {
```
Comment (annaester) [inline code/style.css:65]: Here! hover:hover - it took me some time to understand what it was reffering to.

**Review by annaester — COMMENTED** [review]
Hi Maria!

Such a good job! I am amazed at how much you have learnt. There are parts of your code I don't really understand but I look forward to doing so in the near future. Is fun to read someone elses code and see how they approached the assignment. 

My feedback is that there seem to be some repetition in the HTML. I left some comments in the code. Instead of repeating the ie navbar/button/logo several times for each part in the HTML couldn't the media queries in CSS been used to style it differently? With all the repetition it was a bit difficult to read the code in a swiftly way. 

Also I thought about the lack of text. I noticed in my code that it was a bit challenging to combine cards with image and text and cards with only text in the CSS. Did you consider having more text? 

Great work, it was inspiring to look at.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+The challenge this week was to make the site responsive, that is to write code where the layout and content change depending on if the user visit the site from a mobile, tablet or desktop. 
+
+For this I used a combination of CSS Flexbox and CSS Grid to get the layout I wanted. I also used the mobile first approach where I created the mobile version first and then add CSS for tablet and desktop. 
+
+I found it very useful to test out my ideas on Codepen - and get a working solution that I could test and verify before I addded any new code to the project. 
+
+In addition to learning how to work with Flexbox and Grid, I also studied how I can make use of transition and scaleX to add hover effects to links and buttons.
```
[the remark this answers is not in the document]
Comment (molbimien) [reply README.md:13]: scaleX (and Y) can be used to transform the elements and for example allow you to move, rotate, scale, and skew elements. Learn more on https://www.w3schools.com/css/css3_2dtransforms.asp 

I used scaleX for the hover effect on the menu links and also the button. To make the cards move when you hover - I used the following:

```
.card:hover, .card:focus, .card:active {
    filter: none;
    top: -0.25rem;
    box-shadow: 0 4px 5px rgba(0,0,0,0.2);
}
```

File: code/index.html [hunk 4]
```diff
@@ -4,19 +4,194 @@
...
   <body>
-    <header>
-      <!-- Some kind of logo and navbar -->
-    </header>
-
-    <section class="big-news">
+    <div class="container">
+      <!-- a header wih a site name, a logo and a navbar -->
+      <header class="mobile-only-header">
+        <a href="#" >
```
[the remark this answers is not in the document]
Comment (molbimien) [reply code/index.html:15]: It's just common practice I would assume - to be able to click on the logo to "get home". I just forgot to add the link  😂

File: code/index.html [hunk 5]
```diff
@@ -4,19 +4,194 @@
...
+          <p class="nav"><a class="hover" href="#">Subscribe</a></p>
+        </nav>
+      </header>
+      <header class="desktop-only-header">
+      <a href="#" >
+        <img class="logo" src="./logo.png" alt="Molbimien News" />
+      </a>
+      <h1 class="page-title">This awesome news site</h1>
+      <nav>
+        <p class="nav"><a class="hover" href="#">About</a></p>
```
[the remark this answers is not in the document]
Comment (molbimien) [reply code/index.html:35]: Yes, this is really good input and the last few days I think I've learned a trick or two to do just that. Hopefully I will have some time over to rework this!

File: code/style.css [hunk 6]
```diff
@@ -1,4 +1,322 @@
...
+  margin: 0 0 16em 0;
+}
+
+.mobile-big-news-image {
+  position: absolute;
+  top: 0;
+  left: 0;
+  height: 100vh;
+  max-width: 100%;
+  z-index: -1;
```
[the remark this answers is not in the document]
Comment (molbimien) [reply code/style.css:95]: Yes, that was really useful to get the logo in the mobile to be shown correctly! And also for the button hover effect. :)

File: code/style.css [hunk 7]
```diff
@@ -1,4 +1,322 @@
...
+  top: 0;
+  left: 50%;
+  -webkit-transform: translateX(-50%);
+  transform: translateX(-50%);
+  width: 0;
+  height: 100%;
+  background: #DB398D;
+  z-index: -1;
+  -webkit-transition: width 300ms ease-in-out;
+  transition: width 300ms ease-in-out;
```
[the remark this answers is not in the document]
Comment (molbimien) [reply code/style.css:151]: Thanks! I will make sure to add comments and more information in the code the next time!

File: code/style.css (hunk 2 above)
Comment (molbimien) [reply code/style.css:65]: Yes I can see that. I think I could actually replace this with a:hover - that would make more sense 😂

File: code/style.css [hunk 8]
```diff
@@ -1,4 +1,322 @@
...
+  transition: transform .5s ease;
+}
+
+.hover:before {
+  left: 0;
+  bottom: 0;
+  width: 100%;
+  height: 2px;
+  background: #DB398D;
+  transform: scaleX(0);
```
[the remark this answers is not in the document]
Comment (molbimien) [reply code/style.css:71]: See my reply on one of your other comments 😃

**Pull request closed without merging**